src/vs/code/electron-main/menus.ts

```diff
@@ -242,6 +255,10 @@ export class VSCodeMenu {
 
 	private install(): void {
 
+		// Auto hide menu bar
+		const windows = this.windowsService.getWindows();
+		windows.forEach(w => w.win.setAutoHideMenuBar(this.currentAutoHideMenuBar));
```

Comment:
Is this the right place to put this? 

Feels a bit wasteful to create the whole menu from scratch when I just want to update a property of the window?